index.bs

```diff
@@ -165,6 +169,92 @@ To <dfn abstract-op lt="set-dest">set the `Sec-Fetch-Dest` header</dfn> for a [=
 </div>
 
 
+The `Sec-Fetch-Frame-Ancestors` HTTP Request Header {#sec-fetch-frame-ancestors-header}
+---------------------------------------------------------------------------------------
+
+The <dfn http-header export>`Sec-Fetch-Frame-Ancestors`</dfn> HTTP request header exposes the
+relationship between a [=request=]'s initiator's origin, its target origin, and the origins of
+the chain of ancestor documents that embed the initiator. It is a [=Structured Header=] whose
+value is a [=structured header/token=]. [[!I-D.ietf-httpbis-header-structure]] Its ABNF is:
+
+```
+Sec-Fetch-Frame-Ancestors = sh-token
+```
+
+Valid `Sec-Fetch-Frame-Ancestors` values include "`cross-site`", "`same-origin`", and "`same-site`". In order to support forward-compatibility with as-yet-unknown request types,
+servers SHOULD ignore this header if it contains an invalid value.
+
+<div algorithm="set `Sec-Fetch-Frame-Ancestors`">
+To <dfn abstract-op lt="set-frame-ancestors">set the `Sec-Fetch-Frame-Ancestors` header</dfn> for a [=request=] |r|:
+
+<ol class="algorithm">
+    1.  Assert: |r|'s [=request/url=] is a [=potentially trustworthy URL=].
+
+    1.  If |r|'s [=request/destination=] is "`document`", return.
+
+    1.  Let |header| be a [=Structured Header=] whose value is a [=structured header/token=].
+
+    1.  Set |header|'s value to "`same-origin`"..
+
+    1.  For each |ancestor| in |r|'s [=request/client=]'s [=environment settings object/global object=]'s
+        [=associated Document|associated `Document`=]'s [=Document/inclusive ancestor navigables=]:            
+
+        1.  Let |ancestor origin| be |ancestor|'s [=navigable/active document=]'s [=Document/origin=].
+
+        1.  If |ancestor origin| is not [=same site=] with |r|'s [=request/origin=], set |header|'s value
+            to "`cross-site`", and [=iteration/break=].
+
+        1.  If |ancestor origin| is not [=same origin=] with |r|'s [=request/origin=], set |header|'s value
+            to "`same-site`".
+
+    1.  [=header list/Set a structured field value=]
+        &#96;<a http-header>`Sec-Fetch-Frame-Ancestors`</a>&#96;/|header| in |r|'s [=request/header list=].
```

Comment:
It looks like the backticks got messed up here? You can use `<code>...</code>` instead of backticks if you need to make them work nicely with HTML.

Separately, the phrasing here looks like it'll render strangely, like `Set a structured field value Sec-Fetch-Frame-Ancestors/header in r's header list`. That sentence reads oddly to me, but more importantly, that algorithm accepts a tuple which we don't appear to be giving here. I'd write it as something like: `Set a structured field value given <code>("Sec-Fetch-Frame-Ancestors", |header|)</code> in |r|'s [=request/header list=].`